app/helpers/concerns/with_student_path_navigation.rb

```diff
@@ -10,7 +10,7 @@ def next_lesson_button(guide)
   end
 
   def next_exercise_button(exercise)
-    next_button(exercise) || next_button(exercise.guide.lesson)
+    AdaptiveNavigation.new(self).button(exercise) || next_button(exercise.guide.lesson)
```

Comment:
Can we delegate this to a method like `adaptive_navigation_button`?

Reply:
Done, I think. Check it out?